### PR TITLE
Update zfs operator from upstream

### PR DESCRIPTION
**Pull request by sonicaj**
Upstream openebs csi provisioner has introduced a new CRD which is required for it to function properly. These changes are introduced to catch up with the changes so the CSI can function as desired.

### CODE REVIEW

**Review by rick-mesta — APPROVED** [review]

**Comment by rick-mesta** [comment]
please wait for review from @themylogin before merging 🙏🏼

File: manifests/zfs-operator.yaml
Comment (themylogin) [inline manifests/zfs-operator.yaml:1974]: Why spaces here?

**Review by themylogin — APPROVED** [review]

File: manifests/zfs-operator.yaml
Comment (sonicaj) [reply manifests/zfs-operator.yaml:1974]: I have copied it as it is from upstream, i can fix it but probably would be nice if i made a PR upstream as we are going to diverge otherwise

**Review by themylogin — APPROVED** [review]

**Pull request merged**
